bluecellulab/analysis/inject_sequence.py

```diff
@@ -0,0 +1,130 @@
+"""Module for injecting a sequence of protocols to the cell."""
+from __future__ import annotations
+from enum import Enum, auto
+from typing import NamedTuple, Sequence, Dict
+
+import neuron
+import numpy as np
+from bluecellulab.cell.core import Cell
+from bluecellulab.cell.template import TemplateParams
+from bluecellulab.simulation.simulation import Simulation
+from bluecellulab.stimulus.factory import Stimulus, StimulusFactory
+from bluecellulab.utils import IsolatedProcess
+
+
+class StimulusName(Enum):
+    """Allowed values for the StimulusName."""
+    AP_WAVEFORM = auto()
+    IDREST = auto()
+    IV = auto()
+    FIRE_PATTERN = auto()
+
+
+class Recording(NamedTuple):
+    """A tuple of the current, voltage and time recordings."""
+    current: np.ndarray
+    voltage: np.ndarray
+    time: np.ndarray
+
+
+StimulusRecordings = Dict[str, Recording]
+
+
+def run_stimulus(
+    template_params: TemplateParams,
+    stimulus: Stimulus,
+    section: str,
+    segment: float,
+    duration: float,
+) -> Recording:
+    """Creates a cell and stimulates it with a given stimulus.
+
+    Args:
+        template_params: The parameters to create the cell from a template.
+        stimulus: The input stimulus to inject into the cell.
+        section: Name of the section of cell where the stimulus is to be injected.
```

Comment:
Why now we have to pass the name insteaad of the section object?
If there is no other work-around maybe renaming this param to `section_name` would give more hint of what we need to pass

Reply:
We might as well pass the section object. I thought it would be easier to specify the section e.g. `soma[0]` or `dend[15]`. Do you prefer passing the actual section object?

Reply:
One limitation with passing the sections is that The subprocess will have a different section object and the key (section_name) can serve as the mapping between the sections in different processes.

Reply:
In other words, NEURON sections like `cADpyr_L2TPC_bluecellulab_0x7f23a0d1c190[0].dend[1]` are process dependent and I thought it would be misleading to pass that object to `apply_multiple_step_stimuli`.

Reply:
Ah thanks, sure let me rename it to `section_name`.

Reply:
Great! I've tested and this is working now